src/wallet.cpp

```diff
@@ -1715,22 +1757,21 @@ bool CWallet::GetStakeWeight(uint64_t& nWeight)
 
     vector<const CWalletTx*> vwtxPrev;
 
-    set<pair<const CWalletTx*,unsigned int> > setCoins;
-    int64_t nValueIn = 0;
+    vector<pair<const CWalletTx*,unsigned int> > vCoins;
+    std::string sError = "";
+    nWeight = 0;
 
-    if (!SelectCoinsForStaking(nBalance - nReserveBalance,  GetAdjustedTime(), setCoins, nValueIn))
+    if (!SelectCoinsForStaking(GetAdjustedTime(), vCoins, sError))
         return false;
 
-    if (setCoins.empty())
+    if (vCoins.empty())
```

Comment:
Minor stuff: we can remove this check now that `SelectCoinsForStaking()` returns false when it finds no UTXOs.

Reply:
yes we can its only a few extra cycles anyway

Reply:
perhaps i should check if either are empty and send back the appropriate reason?

```
if (vCoinsRet.empty())
{
    if (fMiner)
        sError = vCoins.empty() ? _("No mature coins") : _("No utxos available due to reserve balance");

    return false;
}

Reply:
lmk i think that would be better instead of no mature coins sending back no utxos available due to reserve

Reply:
This `vCoins` in `GetStakeWeight()` is different from the one in `SelectCoinsForStaking()`. I don't think you need to do that because you're already returning false and setting the reason for `vCoins.empty()` above this:

https://github.com/gridcoin-community/Gridcoin-Research/blob/5f24441671f844fc8f0009b87d16f16116faae0b/src/wallet.cpp#L1531-L1537

That will happen before the function filters the UTXOs, so we should get the right message for both cases as you already wrote it.

Reply:
oh ya so i'll put it back to way it was and remove that check in GetStakeWeight then, correct

Reply:
Yep, that should do it. I think this is finished 🙂

Reply:
thank you.